src/Lykke.Service.KycReports.Services/Reports/KycReportingService.cs

```diff
@@ -0,0 +1,766 @@
+                    {
+                        var kycOfficer = itemsToday.Key.KycOfficer;
+                        var partnerName = itemsToday.Key.PartnerName;
+
+                        var onBoardedCount = itemsToday
+                            .Count(i => (
+                                (i.StatusCurrent == KycStatus.ReviewDone || i.StatusCurrent == KycStatus.Ok) && (i.StatusPrevious == KycStatus.Pending))
+                            );
+
+                        var declinedCount = itemsToday
+                            .Count(i => (i.StatusCurrent == KycStatus.RestrictedArea || i.StatusCurrent == KycStatus.Rejected) && (i.StatusPrevious == KycStatus.Pending));
+
+                        var toResubmitCount = itemsToday
+                            .Count(i => 
+                                (i.StatusCurrent == KycStatus.NeedToFillData && (i.StatusPrevious == KycStatus.Pending))
+                            );
+
+                        var reportRow = new KycOfficerStatsDataReport()
+                        {
+                            ReportDay = startOfDay,
+                            KycOfficer = kycOfficer,
+                            PartnerName = partnerName,
+                            OnBoardedCount = onBoardedCount,
+                            DeclinedCount = declinedCount,
+                            ToResubmitCount = toResubmitCount
+                        };
+
+                        await _reportRepository.InsertRow(reportRow);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    await _log.WriteErrorAsync("SrvReports", "GenerateKycOfficerStats (insert new rows)", startOfDay.ToString(), ex);
+                    isGeneratedOk = false;
+                }
+            }
+
+            return isGeneratedOk;
+        }
+
+        public async Task<bool> RebuildKycOfficerStats()
+        {
+            var startDate = DateTime.Today.AddDays(-14);
+            return await GenerateKycOfficerStats(startDate, false);
+        }
+        
+        private async Task<IEnumerable<string>> GetKycOfficersPerformanceJsonRows(DateTime startDate, DateTime endDate)
+        {
+            startDate = new DateTime(startDate.Year, startDate.Month, startDate.Day);
+
+            //var yesterday = DateTime.Today.AddDays(-1);
+            var today = DateTime.Today;
+            if (endDate > today)
+                endDate = today;
+
+            if (endDate < startDate)
+                return new List<string>();
+
+
+            await GenerateKycOfficersPerformance(startDate);
+            var jsonReport = await _reportRepository.GetKycOfficersPerformanceJsonRows(startDate, endDate);
+
+            return jsonReport.Where(jRow => !jRow.Contains(KycOfficersPerformanceRow.EmptyDayKycOfficer));
+        }
+
+        private async Task<IEnumerable<KycOfficersPerformanceRow>> GenerateKycOfficersPerformance(DateTime from)
```

Comment:
Same issues as in GenerateKycOfficerStats method

Reply:
rewritten